random/weighted_random_list.go

```diff
@@ -0,0 +1,134 @@
+package random
+
+import (
+	"fmt"
+	"hash/fnv"
+	"math/rand"
+	"sort"
+	"time"
+)
+
+//go:generate mockery -all -case=underscore
+
+// Interface to use the Weighted Random
+type WeightedRandomList interface {
+	Get() interface{}
+	GetWithSeed(seed string) (interface{}, error)
+	List() []interface{}
+}
+
+// Interface for items that can be used along with WeightedRandomList
+type Comparable interface {
+	Compare(to Comparable) bool
+}
+
+// Structure of each entry to select from
+type Entry struct {
+	Item   Comparable
+	Weight float32
+}
+
+type internalEntry struct {
+	entry        Entry
+	currentTotal float32
+}
+
+// WeightedRandomList selects elements randomly from the list taking into account individual weights.
+// Weight has to be assigned between 0 and 1.
+// Support deterministic results given a particular seed and sortKey
+type weightedRandomListImpl struct {
+	entries     []internalEntry
+	totalWeight float32
+}
+
+func validateEntries(entries []Entry) error {
+	if len(entries) == 0 {
+		return fmt.Errorf("entries is empty")
+	}
+	for _, entry := range entries {
+		if entry.Weight < 0 || entry.Weight > float32(1) {
+			return fmt.Errorf("invalid weight %f", entry.Weight)
+		}
+	}
+	return nil
+}
+
+// Given a list of entries and sortKey, return WeightedRandomList
+// The sortKey indicates the field in the object to be used for sorting.
```

Comment:
Update docs here since we dropped sortKey.
Also mention the selection criteria (nothing with weights or only ones with weights)